### DISCUSSION TITLE
Why TCP ?

### DISCUSSION THREAD

**Discussion by AdiAbuAli** [mrlt8/docker-wyze-bridge#255] [Q&A]
Hi,
Why we need to use TCP protocol, if the container and NVR run on same platform?
From destination point of view, there will be no packet loss, as both run on same machine.
Also from source to container, TCP is defined between FFMPEG and RTSP_SIMPLE_SERVER, which both run same container even....and FFMPEG doesn't have back feedback for media with Wyze, but just copy the stream.
So why not UDP?

**Comment by mrlt8** [comment]
You can manually change it with env, but from what I remember UDP-in caused issues with TCP-out.

`RTSP_PROTOCOLS=UDP`

**Comment by AdiAbuAli** [comment]
Changing that env protocols=udp is enough?

I tried to change it E2E...

I defined -p 8000:8000 UDP
Protocols=udp
Ffmpeg cmd=......rtsp_transport udp rtsp://0.0.0.0:8000/
But i got error tcp://0.0.0.0:8000 connection refused.

Thanks,
________________________________

**Reply by mrlt8** [reply]
Should be enough:
```
[session 485290439] is publishing to path 'front', 1 track with UDP
```
